app/services/supplementary_data.py

```diff
@@ -78,6 +96,22 @@ def get_supplementary_data(*, dataset_id: str, unit_id: str, survey_id: str) ->
     raise SupplementaryDataRequestFailed
 
 
+def decrypt_supplementary_data(supplementary_data: MutableMapping) -> Mapping:
+    if encrypted_data := supplementary_data.get("data"):
+        try:
+            decrypted_data = JWEHelper.decrypt(
+                encrypted_data, key_store=get_key_store(), purpose=KEY_PURPOSE_SDS
+            )
+            supplementary_data["data"] = json.loads(decrypted_data)
+            return supplementary_data
+        except InvalidTokenException as e:
+            raise InvalidSupplementaryData from e
+
+    raise ValidationError(
+        "Supplementary data response cannot be decrypted without the key 'data'"
```

Comment:
Do we go with this or something like 'No data to decrypt'?

Reply:
Changed message 👍